### PR TITLE
Issues/122 rwh crypto utils for pdq

### LINKED ISSUE #122: PDQ client: replace security implementations with the RWH crypto-utils library
The current implementation of the PDQ client uses duplicated code from the RWH crypto-utils library. 
--> Replace the security package of the PDQ client with the the RWH crypto-utils library.

### LINKED ISSUE #127: Rework .gitignore
Currently multiple `.gitignore` are used, only specifying files that are part of the same directory or module.

We might want to try consolidating all `.gitignore` files into one top level file. This file could contain general rules `**/foo` that apply everywhere and specialized rules for single files in specific locations.

### PR DESCRIPTION
**Pull request by wetret**
- uses crypto utils library in pdq client
- reworks gitignore

closes #122 
closes #127

### CODE REVIEW

File: dsf-mpi/dsf-mpi-client-pdq/src/main/java/org/highmed/mpi/client/message/MessageHelper.java [hunk 1]
```diff
@@ -66,10 +66,19 @@ public QBP_Q21 createPatientDemographicsQuery(String senderApplication, String s
...
+			{
+				terser.set(searchParameter.getLocation(), searchParameter.getValue());
+			}
+			else
+			{
+				terser.set("/QPD-3(" + qpd3counter + ")-1", searchParameter.getField());
+				terser.set("/QPD-3(" + qpd3counter + ")-2", searchParameter.getValue());
+				qpd3counter++;
+			}
 		}
```
Comment (hhund) [inline dsf-mpi/dsf-mpi-client-pdq/src/main/java/org/highmed/mpi/client/message/MessageHelper.java:82]: The qpd3counter variable does not need to be visible outside the loop. I would suggest using a `for (int`... loop.
```java
for (int qpd3counter = 0; qpd3counter < searchParameters.size(); qpd3counter++)
{
	QueryParameter searchParameter = searchParameters.get(qpd3counter);

	if (searchParameter.hasLocation())
	{
		terser.set(searchParameter.getLocation(), searchParameter.getValue());
	}
	else
	{
		terser.set("/QPD-3(" + qpd3counter + ")-1", searchParameter.getField());
		terser.set("/QPD-3(" + qpd3counter + ")-2", searchParameter.getValue());
	}
}
```

File: dsf-tools/dsf-tools-test-data-generator/cert/ca/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Will contain the test ca certificate and privkey generated in the build process
```
Comment (hhund) [inline dsf-tools/dsf-tools-test-data-generator/cert/ca/README.md:1]:
```suggestion
Will contain the Test CA certificate and private-key generated in the build process.

**Make sure to protect the CA private-key from third-party access if you have installed the Test CA certificate in your systems or browsers certificate store.**
```

**Review by hhund — CHANGES_REQUESTED** [review]
create `dsf-bpe/dsf-bpe-server-jetty/docker/lib/README.md` file
remove existing `dsf-bpe/dsf-bpe-server-jetty/docker/lib/.gitignore` file
create `dsf-fhir/dsf-fhir-server-jetty/docker/lib/README.md` file
remove existing `dsf-fhir/dsf-fhir-server-jetty/docker/lib/.gitignore` file

for further remarks see line-by-line comments

**Comment by wetret** [comment]
I made the requested changes.

**Review by hhund — APPROVED** [review]

**Pull request merged**